tests/unittest_nodes.py

```diff
@@ -386,12 +386,19 @@ def test_if_sys_guard() -> None:
         assert isinstance(code, list) and len(code) == 3
 
         assert isinstance(code[0], nodes.If)
-        assert code[0].is_sys_guard() is True
+        with pytest.warns(DeprecationWarning) as records:
+            assert code[0].is_sys_guard() is True
+            assert len(records) == 1
```

Comment:
These to methods will be removed soon, so no real need to assert each Deprecation warning. Why not use a filter decorator on the test functions?

```py
    @staticmethod
    @pytest.mark.filterwarnings("ignore:.*is_sys_guard:DeprecationWarning")
    def test_if_sys_guard() -> None:
        ...

    @staticmethod
    @pytest.mark.filterwarnings("ignore:.*is_typing_guard:DeprecationWarning")
    def test_if_typing_guard() -> None:
        ...
```